src/main/java/space/obminyashka/items_exchange/controller/UserController.java

```diff
@@ -88,11 +90,16 @@ public List<AdvertisementTitleDto> getCreatedAdvertisements(@ApiIgnore Authentic
     @ApiResponses(value = {
             @ApiResponse(code = 202, message = "ACCEPTED"),
             @ApiResponse(code = 400, message = "BAD REQUEST"),
-            @ApiResponse(code = 403, message = "FORBIDDEN")})
+            @ApiResponse(code = 403, message = "FORBIDDEN"),
+            @ApiResponse(code = 409, message = "CONFLICT")})
     @ResponseStatus(HttpStatus.ACCEPTED)
     public String updateUserPassword(@Valid @RequestBody UserChangePasswordDto userChangePasswordDto,
-                                     @ApiIgnore Authentication authentication) throws InvalidDtoException {
+                                     @ApiIgnore Authentication authentication)
+            throws InvalidDtoException, DataConflictException {
         User user = findUserByValidCredentials(authentication, userChangePasswordDto.getOldPassword());
+        if (!patternMatches(user.getEmail(), PatternHandler.EMAIL)) {
+            throw new DataConflictException(getMessageSource("invalid.email"));
+        }
```

Comment:
Didn't get that part. 
Am I right that for the method "updateUserPassword" you validate the email of an already saved user?